tests/test.py

```diff
@@ -0,0 +1,30 @@
+import pytest
+import pycurl
+import io
+import json
+
+
+def test_upload():
```

Comment:
this does more than testing the upload, you're also testing an image and comparing it against an assumed value. suggest renaming this test to reflect what it's actually doing.

You could even split this up into something like the following and add more tests to test for errors (like a 400 for an invalid file format :) )

```
test_file = 'static/img/images/7-action-athlete-896567.jpg'
temp_file = 'static/img/images/MAX-7-action-athlete-896567.jpg'
baseball = 'a baseball player swinging a bat at a ball'

def baseTest():
    c = pycurl.Curl()
    b = io.BytesIO()
    c.setopt(pycurl.URL, 'http://localhost:8088/upload')
    c.setopt(pycurl.HTTPHEADER,
             ['Accept:application/json', 'Content-Type: multipart/form-data'])

def test_upload_baseball(baseTest):
    c.setopt(pycurl.HTTPPOST, [('file', (pycurl.FORM_FILE, test_file))])
    c.setopt(pycurl.WRITEFUNCTION, b.write)
    c.perform()
    assert c.getinfo(pycurl.RESPONSE_CODE) == 200
    c.close()
    response = b.getvalue()
    response = json.loads(response)[0]
    assert response['file_name'] == temp_file
    assert response['caption'] == baseball
```

Reply:
I'll keep this in mind when updating to use requests